### DISCUSSION TITLE
Split compiler in two

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2847] [Suggestions]
Compiler should not be validating code. It should just be emitting lua functions to run. There needs to be an analyzer step which validates your code and emits a typed AST / HIR.

This will open up the door to a new optimizer and faster validation times (not wasting your time on codegen)

**Comment by bjcscat** [comment]
wireil

**Reply by vurvdev** [reply]
All this would do is complicate things right now so I have no intention of going through with it or accepting it into wire. ZCPU wouldn't use it, and there's only E2 at the moment so
